src/common/lru_cache.h

```diff
@@ -453,6 +461,208 @@ std::shared_ptr<CacheMetrics>
+    if (ll_.begin() != ll_.end()) {
+        *eliminated = ll_.back().value;
+        RemoveElement(--ll_.end());
+        return true;
+    }
+    return false;
+}
+
+template <typename K,  typename V, typename KeyTraits, typename ValueTraits>
+void TimedLRUCache<K, V, KeyTraits, ValueTraits>::MoveToFront(
+    const typename std::list<Item>::iterator &elem) {
+    Item duplica{elem->key, elem->value, elem->time};
+    ll_.erase(elem);
+    ll_.push_front(duplica);
+    cache_[*(duplica.key)] = ll_.begin();
+}
+
+template <typename K,  typename V, typename KeyTraits, typename ValueTraits>
+void TimedLRUCache<K, V, KeyTraits, ValueTraits>::RemoveLocked(const K &key) {
+    auto iter = cache_.find(key);
+    if (iter != cache_.end()) {
+        RemoveElement(iter->second);
+    }
+}
+
+template <typename K,  typename V, typename KeyTraits, typename ValueTraits>
+bool TimedLRUCache<K, V, KeyTraits, ValueTraits>::PutLocked(
+    const K &key, const V &value, V *eliminated) {
+    auto iter = cache_.find(key);
+    // delete the old value if already exist
+    if (iter != cache_.end()) {
+        RemoveElement(iter->second);
+    }
+
+    // put new value
+    uint64_t now = TimeUtility::GetTimeofDaySec();
+    Item kv{nullptr, value, now};
+    ll_.push_front(kv);
+    cache_[key] = ll_.begin();
+    ll_.begin()->key = &(cache_.find(key)->first);
+    if (cacheMetrics_ != nullptr) {
+        cacheMetrics_->UpdateAddToCacheCount();
+        cacheMetrics_->UpdateAddToCacheBytes(
+            KeyTraits::CountBytes(key)  + ValueTraits::CountBytes(value) +
+            CacheTraits<uint64_t>::CountBytes(now));
+    }
+    if (maxCount_ != 0 && ll_.size() > maxCount_) {
+        return RemoveOldest(eliminated);
+    }
+    return false;
+}
+
+template <typename K,  typename V, typename KeyTraits, typename ValueTraits>
+std::shared_ptr<CacheMetrics>
+    TimedLRUCache<K, V, KeyTraits, ValueTraits>::GetCacheMetrics() const {
+    return  cacheMetrics_;
+}
+
+template <typename K,  typename V, typename KeyTraits, typename ValueTraits>
+uint64_t TimedLRUCache<K, V, KeyTraits, ValueTraits>::Size() {
+    ::curve::common::WriteLockGuard guard(lock_);
+    return cache_.size();
+}
+
+template <typename K,  typename V, typename KeyTraits, typename ValueTraits>
+void TimedLRUCache<K, V, KeyTraits, ValueTraits>::Put(
+    const K &key, const V &value) {
+    V eliminated;
+    ::curve::common::WriteLockGuard guard(lock_);
+    PutLocked(key, value, &eliminated);
+}
+
+template <typename K,  typename V, typename KeyTraits, typename ValueTraits>
+bool TimedLRUCache<K, V, KeyTraits, ValueTraits>::Put(
+    const K &key, const V &value, V *eliminated) {
+    ::curve::common::WriteLockGuard guard(lock_);
+    return PutLocked(key, value, eliminated);
+}
+
+template <typename K,  typename V, typename KeyTraits, typename ValueTraits>
+bool TimedLRUCache<K, V, KeyTraits, ValueTraits>::Get(const K &key, V *value) {
```

Comment:
Compared with LRUCache, I understand that there is an additional judgment of IsTimeout, whether it can be directly modified in the original LRUCache？

Reply:
The main reason is if add the timeout attr in item of LRUCache will increase memory used for the scenes don't need timeout.